internal/storage/storage.go

```diff
@@ -18,13 +18,18 @@ const (
 // EvaluationRule represents a rule and constraints required for evaluating if a
 // given flagKey matches a segment
 type EvaluationRule struct {
-	ID               string                 `json:"id"`
-	NamespaceKey     string                 `json:"namespace_key,omitempty"`
-	FlagKey          string                 `json:"flag_key,omitempty"`
-	SegmentKey       string                 `json:"segment_key,omitempty"`
-	SegmentMatchType flipt.MatchType        `json:"segment_match_type,omitempty"`
-	Rank             int32                  `json:"rank,omitempty"`
-	Constraints      []EvaluationConstraint `json:"constraints,omitempty"`
+	ID              string
```

Comment:
we cant remove the JSON tags added in #1910 as they are required for caching, we also need to add them to `EvaluationSegment`

Reply:
Think it might have been a bad merge.